src/err/mod.rs

```diff
@@ -225,18 +225,30 @@ impl PyErr {
         PyErr::from_state(state)
     }
 
+    /// Deprecated form of [`PyErr::get_type_bound`].
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`PyErr::get_type` will be replaced by `PyErr::get_type_bound` in a future PyO3 version"
+        )
+    )]
+    pub fn get_type<'py>(&'py self, py: Python<'py>) -> &'py PyType {
+        self.get_type_bound(py).into_gil_ref()
+    }
+
     /// Returns the type of this exception.
     ///
     /// # Examples
     /// ```rust
-    /// use pyo3::{exceptions::PyTypeError, types::PyType, PyErr, Python};
+    /// use pyo3::{exceptions::PyTypeError, types::PyType, PyErr, Python, prelude::PyAnyMethods};
```

Comment:
Generally I've been preferring just importing the prelude in user-facing code:

```suggestion
    /// use pyo3::{prelude::*, exceptions::PyTypeError, types::PyType};
```